public/app/plugins/panel/alertlist/UnifiedAlertList.tsx

```diff
@@ -58,14 +61,16 @@ export function UnifiedAlertList(props: PanelProps<UnifiedAlertListOptions>) {
   });
 
   useEffect(() => {
-    dispatch(fetchAllPromRulesAction());
-    const sub = dashboard?.events.subscribe(TimeRangeUpdatedEvent, () => dispatch(fetchAllPromRulesAction()));
+    //we need promRules and rulerRules for getting the uid when creating the alert link in panel in case of being a rulerRule.
+    dispatch(fetchAllPromAndRulerRulesAction());
+    const sub = dashboard?.events.subscribe(TimeRangeUpdatedEvent, () => dispatch(fetchAllPromAndRulerRulesAction()));
     return () => {
       sub?.unsubscribe();
     };
   }, [dispatch, dashboard]);
 
   const promRulesRequests = useUnifiedAlertingSelector((state) => state.promRules);
+  const combinedRules = useCombinedRuleNamespaces();
 
   const dispatched = rulesDataSourceNames.some((name) => promRulesRequests[name]?.dispatched);
   const loading = rulesDataSourceNames.some((name) => promRulesRequests[name]?.loading);
```

Comment:
I think using `promRulesRequests` for `dispatched` and `loading` state is no longer valid, because ruler rules can still be in progress. We should probably use something similar to what we have on the alert list page?

Reply:
💯